### ISSUE TITLE
Sad sprite rotation inaccuracy

### ISSUE THREAD

**Issue by kaesesahnetorte** [pyglet/pyglet#486]
I think I discovered a bug concerning sprite rotation.

I inserted a sprite with 270° rotation
but it shows weird artifacts on its borders, as if it was not rotated all the way.

I added my simplified code base for recreating this bug:
[sprite_bug.zip](https://github.com/pyglet/pyglet/files/7318385/sprite_bug.zip)

**Comment by caffeinepills** [comment]
Looks like linear interpolation to me, which is expected behavior. (Especially when rotating in non-90 degree intervals.)

If you don't want linear you can change it to nearest by doing the following before loading any textures:

```py
pyglet.image.Texture.default_min_filter = pyglet.gl.GL_NEAREST
pyglet.image.Texture.default_mag_filter = pyglet.gl.GL_NEAREST
```

**Comment by benmoran56** [comment]
The rotation math can be found here: 
https://github.com/pyglet/pyglet/blob/pyglet-1.5-maintenance/pyglet/sprite.py#L402
It looks like it's probably just lack of float precision... 
If you set the image anchor to the center, it still maintains enough precision to appear as expected. 

Perhaps we could replace this with some Matrix math and see if it makes a difference.

**Comment by benmoran56** [comment]
Closing for now since workarounds exist. We can't really do anything to fix the float precision on the 1.5 branch anyway, but once 2.0 is out for release it will no longer be an issue. 
Thanks!
